test/testcafe/framework/page-objects/UiDemoPageObject.js

```diff
@@ -0,0 +1,33 @@
+import BasePageObject from './BasePageObject';
+import { ClientFunction } from 'testcafe';
+
+
+export default class UiDemoPageObject extends BasePageObject {
+  constructor(t) {
+    super(t);
+  }
+
+  async scrollToNextPage() {
+    const getClientHeight = ClientFunction(() => window.document.documentElement.clientHeight);
+    const windowheight = await getClientHeight();
+    await this.t.scrollBy(0, windowheight);
+  }
+
+  async hideProgressBar() {
+    await ClientFunction(() => {
+      // the testcafe progress bar element is in the shadow dom so we cannot query it directly
+      const style = document.createElement('style');
+      style.textContent = '.progress-bar-hammerhead-shadow-ui { visibility: hidden !important; }';
+      document.head.appendChild(style);
+    })();
+  }
+
+  async stopSpinnerAnimation() {
+    await ClientFunction(() => {
+      const spinnerRoot = document.querySelector('.MuiCircularProgress-root');
+      const spinnerCircle = document.querySelector('.MuiCircularProgress-circle');
+      spinnerRoot.setAttribute('style', 'animation: none;');
+      spinnerCircle.setAttribute('style', 'animation: none;');
+    })();
+  }
```

Comment:
This is a good idea to halt the spinner or put it into a predictable state, but I don't actually see the spinner at all in the screenshots right now.

Reply:
fixed by appending the style to the `head` instead